### DISCUSSION TITLE
Minimize the "overhead" of an additional orm store/ keep pinias simplicity

### DISCUSSION THREAD

**Discussion by svenhue** [CodeDredd/pinia-orm#9] [Ideas]
Hey, so i was thinking about how we can intregrate orm functionality into an pinia store without this much additional states and with the ability to use the pinia actions, getters etc on the state of the models (our tables).

so at this approach we would add a model instance (the plugin), an object of fields and a object named table to our native pinia store and thats all!
Note that the plugin will instantiate the model with the fields of our pinia store
that means we can access our orm through store.model.insert  (store = useStore() of pinia)

Currently im adding the common orm functionality, in my opinion we should focus on adding ONLY orm functionality within this plugin - no additional stores etc..

a store would look like this:

![image](https://user-images.githubusercontent.com/83905274/163692978-afc04984-f2b4-48e2-9b12-289c56daafff.png)

i will add a link to my repo as soon as i can provide a matured code base :)

**Comment by CodeDredd** [comment]
Sounds interesting. But wouldn't that mean to get rid of the current `model` sturcture used in vuex-orm ? I get your point but this solution looks like we lose a bit in structure. It would make sense to include somehow there also old models and haveing this.

Between having multiple stores or just one should be an plugin option so that the user can choose between this. I had that idea also at refactoring of the code.

**Reply by svenhue** [reply]
But what is the way to go here when we want add actions to our orm store? Right now I have to import the repository and use the orm methods. Isn't there a way to write some method which do some query and use it in multiple components like with pinia actions ? 

Or am I missing something here ?

A possible solution could be import the repo in an pinia store?(nested stores are described in the pinia docs)

**Reply by CodeDredd** [reply]
Well my idea was to include the pinia-store configuration to the `model`. That thay each store can have custom actions and more
